### PR TITLE
Issue #10158 - tests to prove out XML only based deployment technique

### LINKED ISSUE #10158: Deploying on Jetty 12 using context XML files will only work when a .properties file with the EE details is also present
**Jetty version(s)**
Jetty 12 (beta 3)

**Java version/vendor** `(use: java -version)`
openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment (build 17.0.7+7-Ubuntu-0ubuntu122.04.2)
OpenJDK 64-Bit Server VM (build 17.0.7+7-Ubuntu-0ubuntu122.04.2, mixed mode, sharing)

**OS type/version**
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy

**Description**
When I follow the steps in https://eclipse.dev/jetty/documentation/jetty-12/operations-guide/index.html#og-deploy-jetty the presence of the Jetty context XML file has no impact - Nothing gets deployed at all.

The (almost) same context XML file has been working for many previous Jetty versions (actually, all the way to Jetty 6.x), I hence believe this is a regression.

**How to reproduce?**
1. Create a Jetty context XML file, similar to the below - I created a file `webapps/simple-war.xml` with contents:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://www.eclipse.org/jetty/configure_10_0.dtd">

<Configure class="org.eclipse.jetty.ee10.webapp.WebAppContext">
  <Set name="contextPath">/simple-war</Set>
  <Set name="war">/home/hyper-v/cargo/core/samples/java/target/deployables-jakarta-ee/simple-war.war</Set>
</Configure>
```

2. Start the Jetty server, the logs show no deployment of the context XML file:
```
2023-07-26 20:38:01.258:INFO :oejs.Server:main: jetty-12.0.0.beta3; built: 2023-07-03T21:07:40.128Z; git: 3c948df481846fa922f841aa20857f18c92736e5; jvm 17.0.7+7-Ubuntu-0ubuntu122.04.2
2023-07-26 20:38:01.295:INFO :oejdp.ScanningAppProvider:main: Deployment monitor ee10 in [file:///home/hyper-v/cargo/core/samples/java/target/jetty-base/webapps/] at intervals 0s
2023-07-26 20:38:01.310:INFO :oejs.DefaultSessionIdManager:main: Session workerName=node0
2023-07-26 20:38:01.331:INFO :oejs.AbstractConnector:main: Started ServerConnector@36b4fd29{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2023-07-26 20:38:01.339:INFO :oejs.Server:main: Started oejs.Server@ea6147e{STARTING}[12.0.0.beta3,sto=5000] @768ms
```

**Other observations - The same with the WAR file copied over**
Copying over the WAR file works:
1. Execute (in my case): `cp /home/hyper-v/cargo/core/samples/java/target/deployables-jakarta-ee/simple-war.war webapps/`
2. Start the Jetty server, now the logs show the WAR is getting deployed:
```
2023-07-26 20:36:37.218:INFO :oejs.Server:main: jetty-12.0.0.beta3; built: 2023-07-03T21:07:40.128Z; git: 3c948df481846fa922f841aa20857f18c92736e5; jvm 17.0.7+7-Ubuntu-0ubuntu122.04.2
2023-07-26 20:36:37.252:INFO :oejdp.ScanningAppProvider:main: Deployment monitor ee10 in [file:///home/hyper-v/cargo/core/samples/java/target/jetty-base/webapps/] at intervals 0s
2023-07-26 20:36:37.261:INFO :oejd.DeploymentManager:main: addApp: App@38cee291[ee10,null,/home/hyper-v/cargo/core/samples/java/target/jetty-base/webapps/simple-war.war]
2023-07-26 20:36:37.365:INFO :oejew.StandardDescriptorProcessor:main: NO JSP Support for /simple-war, did not find org.eclipse.jetty.ee10.jsp.JettyJspServlet
2023-07-26 20:36:37.378:INFO :oejsh.ContextHandler:main: Started oeje10w.WebAppContext@48b67364{Simple Webapp,/simple-war,b=file:///tmp/jetty-0_0_0_0-8080-simple-war_war-_simple-war-any-4096158269517142805/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@189cbd7c{STARTED}}{/home/hyper-v/cargo/core/samples/java/target/jetty-base/webapps/simple-war.war}
2023-07-26 20:36:37.387:INFO :oejes.ServletContextHandler:main: Started oeje10w.WebAppContext@48b67364{Simple Webapp,/simple-war,b=file:///tmp/jetty-0_0_0_0-8080-simple-war_war-_simple-war-any-4096158269517142805/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@189cbd7c{STARTED}}{/home/hyper-v/cargo/core/samples/java/target/jetty-base/webapps/simple-war.war}
2023-07-26 20:36:37.390:INFO :oejs.DefaultSessionIdManager:main: Session workerName=node0
2023-07-26 20:36:37.404:INFO :oejs.AbstractConnector:main: Started ServerConnector@ff55d30{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2023-07-26 20:36:37.415:INFO :oejs.Server:main: Started oejs.Server@ea6147e{STARTING}[12.0.0.beta3,sto=5000] @957ms
```

### PR DESCRIPTION
**Pull request by joakime**
Adding test case to `DistributionTests` to show that an XML based deployment (with the WAR outside of the `${jetty.base}/webapps/` directory) is possible.

### CODE REVIEW

**Review by janbartel — APPROVED** [review]
If it builds, it's fine.

File: tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 1]
```diff
@@ -1529,4 +1530,85 @@ public void testRangeRequestMultiPartRangeResponse(String env) throws Exception
...
+        };
+
+        try (JettyHomeTester.Run runConfig = distribution.start(argsConfig))
+        {
+            assertTrue(runConfig.awaitFor(START_TIMEOUT, TimeUnit.SECONDS));
+            assertEquals(0, runConfig.getExitValue());
+
+            String[] argsStart = {
+                "jetty.http.port=" + httpPort,
+                "jetty.httpConfig.port=" + httpPort
```
Comment (sbordet) [inline tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:1559]: This property `jetty.httpConfig.port` does not exist, remove it.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java (hunk 1 above)
Comment (joakime) [reply tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:1559]: This property is present in many other places as well.
I'll make a different PR to correct all of the deprecated properties.

Comment (joakime) [reply tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java:1559]: Opened Issue #10334

**Review by sbordet — APPROVED** [review]

**Pull request merged**